definitions/checks/system_registration.rb

```diff
@@ -0,0 +1,29 @@
+class Checks::SystemRegistration < ForemanMaintain::Check
+  metadata do
+    label :verify_self_registration
+    description 'Check whether system is self-registered or not'
+    tags :default
+
+    confine do
+      file_exists?('/etc/rhsm/rhsm.conf')
+    end
+  end
+
+  def run
+    if system_is_self_registerd?
+      raise ForemanMaintain::Error::Warn, 'System is self registered'
+    else
+      puts 'System is not self registered'
+    end
+  end
+
+  private
+
+  def system_is_self_registerd?
+    rhsm_hostname.include?(hostname)
```

Comment:
We should match only the last word from the line and match on equality, rather than include: you can have `sat.example.com' registered to `another-sat.example.com` and this check would fail: a corner case, but still…